src/Features/Core/Portable/Diagnostics/Analyzers/IDocumentDiagnosticAnalyzer.cs

```diff
@@ -12,10 +11,14 @@ namespace Microsoft.CodeAnalysis.Diagnostics
     /// </summary>
     internal abstract class DocumentDiagnosticAnalyzer : DiagnosticAnalyzer
     {
+        // REVIEW: why DocumentDiagnosticAnalyzer doesn't have span based analysis?
```

Comment:
Because the point is to analyze the entire document :)

This is the extension point for other languages (like TS) to plugin and do their analysis of the document.  How would we pick the spans with which ot analyze?